pkg/auth/signatures.go

```diff
@@ -0,0 +1,67 @@
+package auth
+
+import (
+	"io"
+	"os"
+
+	"github.com/square/p2/pkg/util"
+	"golang.org/x/crypto/openpgp"
+)
+
+// Something that can verify PGP signatures
+type SigVerifier interface {
+	CheckDetachedSignature(signed, signature io.Reader) (*openpgp.Entity, error)
+}
+
+// A SignatureVerifier that reads its keyring from a file. The file is
+// read every time it is needed.
```

Comment:
Well, currently the signatures for *all* manifests in the intent store are verified [every 5 minutes or when any manifests are scheduled](https://github.com/square/p2/blob/master/pkg/preparer/orchestrate.go#L150). So you're guaranteed to read this file, which could easily have a thousand keys in it, at least 5-10 times every 5 minutes. No, this isn't a dealbreaker, but yes, it is inefficient :)